plugins/by-name/luasnip/default.nix

```diff
@@ -161,49 +365,41 @@ in
     };
   };
 
-  config =
+  extraConfig =
+    cfg:
     let
       loaderConfig =
-        trivial.pipe
+        lib.pipe
           {
             vscode = cfg.fromVscode;
             snipmate = cfg.fromSnipmate;
             lua = cfg.fromLua;
           }
           [
-            # Convert loader options to [{ name = "vscode"; loader = ...; }]
-            (attrsets.mapAttrsToList (name: loaders: lists.map (loader: { inherit name loader; }) loaders))
-            lists.flatten
-
-            (lists.map (
+            (lib.mapAttrsToList (name: loaders: map (loader: { inherit name loader; }) loaders))
+            lib.flatten
+            (map (
               pair:
               let
                 inherit (pair) name loader;
-                options = attrsets.getAttrs [
+                options = lib.getAttrs [
                   "paths"
                   "exclude"
                   "include"
                 ] loader;
               in
               ''
-                require("luasnip.loaders.from_${name}").${optionalString loader.lazyLoad "lazy_"}load(${lib.nixvim.toLuaObject options})
+                require("luasnip.loaders.from_${name}").${lib.optionalString loader.lazyLoad "lazy_"}load(${toLuaObject options})
```

Comment:
IDK how important this is, but it feels like it may be safer to escape `name` by passing it through `toJSON` or "toLuaObject`?

```nix
/*lua*/ ''
  loaders["from_" .. ${toLuaObject name}]
  -- or
  loaders[${toLuaObject "from_${name}"}]
  -- or
  loaders[${toLuaObject ("from_" + name)}]
''
```

Reply:
I don't think this is really needed, imo.